runtime/fastly/builtins/fetch/request-response.cpp

```diff
@@ -2285,7 +2284,13 @@ bool Response::is_upstream(JSObject *obj) {
 
 bool Response::is_grip_upgrade(JSObject *obj) {
   MOZ_ASSERT(is_instance(obj));
-  return JS::GetReservedSlot(obj, static_cast<uint32_t>(Slots::IsGripUpgrade)).toBoolean();
+  return JS::GetReservedSlot(obj, static_cast<uint32_t>(Slots::GripUpgradeRequest)).toInt32() != 0;
+}
+
+host_api::HttpReq Response::grip_upgrade_request(JSObject *obj) {
```

Comment:
You could combine these two calls into a single `std::option<host_api::HttpReq> Response::grip_upgrade_request(JSObject *obj)` instead of needing two separate function calls.

If changing to support UndefinedValue per the next comment, then you'd also need to use an `isUndefined()` for this.